app/controllers/TpsController.scala

```diff
@@ -100,7 +108,7 @@ class TpsController @Inject() (actions:        Actions,
 
     val f = for {
       a <- tpsRepo.findByPcipalSessionId(request.body.PCIPalSessionId)
-      _ = maybeSendEmails(a)
+      _ = maybeSendEmail(a, request.body.Status)
```

Comment:
You can put the status as guard condition here

val f = for {
      a <- tpsRepo.findByPcipalSessionId(request.body.PCIPalSessionId)
      _  if(status == Varified)= maybeSendEmails(a.emailAddress)